packages/core/src/providers/starknet/model.ts

```diff
@@ -1,16 +1,15 @@
 import { defaultProvider, ProviderInterface } from 'starknet'
+import { Connector } from '../../connectors'
 
 export interface StarknetState {
   account?: string
-  hasStarknet: boolean
-  connectBrowserWallet: () => void
+  connect: (connector: Connector) => void
```

Comment:
Should we add a `connectors: Array<Connector>` like wagmi does?

Reply:
[wagmi is a bit different](https://wagmi.sh/guides/connect-wallet#step-2-display-wallet-options) in that you provide `connectors: Array<Connector>` to the `Provider` and then can use connect through `useConnect`, which passes them back with a `connect` function with the same signature as above.

In `starknet-react`, we return `connect` on the main hook, which i like and seems simpler.

I'm not sure why wagmi decided to plub connectors through the provider like they have, maybe they have a good justification. Either way, I would say we should start here to avoid making this change too complex

Reply:
Agree. On further thoughts it's also silly to add an API for that now when there is only one connector.